### ISSUE TITLE
freeipa.ansible_freeipa module fails linter, passed arguments

### ISSUE THREAD

**Issue by mohamedhabas11** [ansible/ansible-lint#4332]
##### Summary

ansible-lint returns an error when passing a default value or a variable to module, and reports the need to match one of the predefined arguments.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
```

```console
ansible-lint 24.9.0 using ansible-core:2.17.4 ansible-compat:24.9.0 ruamel-yaml:0.18.6 ruamel-yaml-clib:0.2.8
```

macOS: Sonoma 14.5
- ansible installation method: one of source, pip, OS package
Brew Cellar
- ansible-lint installation method: one of source, pip, OS package
pip

##### STEPS TO REPRODUCE

```console
ansible-galaxy collection install freeipa.ansible_freeipa
```

```yaml
freeipa_users:
  testuser:
    present: yes 
    realname: "Test User"
    authtype: 
      - "password"
```

```console (paste below)
---
- block:
    - name: "Ensure IPA users are present"
      freeipa.ansible_freeipa.ipauser:
        ipaadmin_password: "{{ freeipa_server_admin_password }}"
        name: "{{ item.key }}"
        userauthtype: "{{ item.value.authtype | d(['password']) }}" #<-- fails the linter
        state: "present"
      loop: "{{ freeipa_users | dict2items }}"

    - name: "Ensure freeipa_users state (Enable/Disable)" # passes
      freeipa.ansible_freeipa.ipauser:
        ipaadmin_password: "{{ freeipa_server_admin_password }}"
        name: "{{ item.key }}"
        state: "{{ item.value.present | ternary ('enabled','disabled') }}"
      loop: "{{ freeipa_users | dict2items }}"
  become: true
```

##### Desired Behavior

passing the linter
Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

Please give some details of what is happening. Include a [minimum complete
verifiable example] with:

- minimized playbook to reproduce the error
- the output of running ansible-lint including the command line used
- if you're getting a stack trace, also the output of
  `ansible-playbook --syntax-check playbook`

```paste below
MohammedH@mohammeds-MacBook-Pro-2 ~/r/ansible (master)> ansible-lint roles/freeipa/tasks/manage_users.yml
WARNING  Listing 1 violation(s) that are fatal
args[module]: value of userauthtype must be one or more of: password, radius, otp, pkinit, hardened, idp, . Got no match for: {{ item.value.authtype | d(['password']) }}
roles/freeipa/tasks/manage_users.yml:3 Task/Handler: Ensure IPA users are present
Read documentation for instructions on how to ignore specific rule violations.

             Rule Violation Summary              
 count tag          profile rule associated tags 
     1 args[module]         syntax, experimental 


```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by alisonlhart** [comment]
We can reproduce this bug. We would appreciate a pull request or contribution to resolve it!
